src/rgw/rgw_admin.cc

```diff
@@ -1030,7 +1030,11 @@ void set_quota_info(RGWQuotaInfo& quota, int opt_cmd, int64_t max_size, int64_t
 
     case OPT_QUOTA_SET:
       if (have_max_objects) {
-        quota.max_objects = max_objects;
+        if (max_objects < 0) {
+          quota.max_objects = -1;
+        } else {
+          quota.max_objects = max_objects;
+        }
```

Comment:
Nothing as such. I just thought it would be same for `max_objects` as we are handling for `max_size`.